docs/user/connections.rst

```diff
@@ -14,9 +14,44 @@ ITargetConnection
     :undoc-members:
     :show-inheritance:
 
-SocketConnection
-================
-.. autoclass:: boofuzz.SocketConnection
```

Comment:
Keep `SocketConnection` in the docs until we completely remove it.
Move it to the end of the list and add a notice about the depreciation.

Reply:
Unfortunately sphinx doesn't have a block for notices, so I settled for a warning.